include/deal.II/grid/grid_tools.h

```diff
@@ -748,6 +748,28 @@ namespace GridTools
                                  const std::vector<bool>      &marked_vertices = std::vector<bool>());
 
   /**
+   * A version of the previous function that exploits an already existing
+   * map between vertices and cells, constructed using the function
+   * GridTools::vertex_to_cell_map, a map of vertex_to_cell_centers, obtained
+   * through GridTools::vertex_to_cells_center_directions, and a guess `cell_hint`.
+   *
+   * @author Luca Heltai, Rene Gassmoeller, 2017
+   */
+  template <int dim, template <int, int> class MeshType, int spacedim>
```

Comment:
what is the reason for using ``int dim, template <int, int> class MeshType, int spacedim`` instead of ``class MeshType``?